### PR TITLE
[refactor](cloud) Extract SyncPoint to common cpp

### PR DESCRIPTION
**Pull request by platoneko**
## Proposed changes

Extract SyncPoint to common cpp directory. Some code shared between BE and Cloud will gradually be extracted to the `$repo_root/common/cpp` directory in the future.

### CODE REVIEW

File: common_cpp/sync_point.h [hunk 1]
```diff
@@ -259,15 +259,12 @@ auto try_any_cast_ret(std::vector<std::any>& any) {
...
-namespace doris::config {
-extern bool enable_injection_point;
-}
-# define TEST_INJECTION_POINT(x) if (doris::config::enable_injection_point) { LOG_INFO("enter inject point {}", x); SYNC_POINT(x); }
-# define TEST_IDX_INJECTION_POINT(x, index) if (doris::config::enable_injection_point) { LOG_INFO("enter inject point {}", x); IDX_SYNC_POINT(x, index); }
-# define TEST_INJECTION_POINT_CALLBACK(x, ...) if (doris::config::enable_injection_point) { LOG_INFO("enter inject point {}", x); SYNC_POINT_CALLBACK(x, __VA_ARGS__); }
-# define TEST_INJECTION_POINT_SINGLETON() if (doris::config::enable_injection_point) { LOG_INFO("enter inject point {}", x); SYNC_POINT_SINGLETON(); }
-# define TEST_INJECTION_POINT_RETURN_WITH_VALUE(x, default_ret_val, ...) if (doris::config::enable_injection_point) { LOG_INFO("enter inject point {}", x); SYNC_POINT_RETURN_WITH_VALUE(x, default_ret_val, __VA_ARGS__); }
-# define TEST_INJECTION_POINT_RETURN_WITH_VOID(x, ...) if (doris::config::enable_injection_point) { LOG_INFO("enter inject point {}", x); SYNC_POINT_RETURN_WITH_VOID(x, __VA_ARGS__); }
+# define TEST_INJECTION_POINT(x) SYNC_POINT(x);
```
Comment (gavinchou) [inline common_cpp/sync_point.h:262]: it seems all injection points are not functioning any more?

File: common_cpp/CMakeLists.txt
Comment (gavinchou) [inline common_cpp/CMakeLists.txt:1]: name the folder "common" or "common/cpp" or "cpp-common"?  
and add a README.md to it.

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: common_cpp/CMakeLists.txt
Comment (w41ter) [reply common_cpp/CMakeLists.txt:1]: How about naming it as `common/rock` and setting the include directory to `common`, then the user would locate and identify the included files by `#include "rock/sync_point.h"`.

File: common_cpp/sync_point.h (hunk 1 above)
Comment (platoneko) [reply common_cpp/sync_point.h:262]: Injection points will be enabled via http api

File: be/CMakeLists.txt [hunk 2]
```diff
@@ -118,6 +118,7 @@ set(BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 set(ENV{DORIS_HOME} "${BASE_DIR}/..")
 set(BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}")
 set(GENSRC_DIR "${BASE_DIR}/../gensrc/build/")
+set(COMMON_CPP_DIR "${BASE_DIR}/../common_cpp/")
```
Comment (gavinchou) [inline be/CMakeLists.txt:121]: common/cpp/xxxx.h

#inlucde "common/cpp/xxx.h"
#include "cpp/xxx.h"

INC_DIR="./common"
INC_DIR="./"

**Comment by platoneko** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by ByteYue — APPROVED** [review]

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**